### PR TITLE
[docs] Clarify purpose of reproducibility docs

### PR DESCRIPTION
**Pull request by stevhliu**
From a quick glance, it seems like the Reusing seeds for deterministic generation and Reproducibility docs are talking about using the same functionality, setting a seed to reproduce results. However, the purposes of these docs are very different. The former is focused on how to improve an image, while the latter is more about testing. This PR clarifies the purpose of these two docs so it's clear what users can expect, and why they should read one versus the other.

### CODE REVIEW

**Review by sayakpaul — APPROVED** [review]
Very nice clean-up, this!

**Review by patrickvonplaten — APPROVED** [review]
Very nice - thanks for the changes

File: docs/source/en/using-diffusers/reproducibility.mdx [hunk 1]
```diff
@@ -45,11 +45,13 @@ image = ddim(num_inference_steps=2, output_type="np").images
...
-value of 1495.1768. What is going on here? Every time the pipeline is run, gaussian noise 
-is created and step-wise denoised. To create the gaussian noise with [`torch.randn`](https://pytorch.org/docs/stable/generated/torch.randn.html), a different random seed is taken every time, thus leading to a different result.
-This is a desired property of diffusion pipelines, as it means that the pipeline can create a different random image every time it is run. In many cases, one would like to generate the exact same image of a certain 
-run, for which case an instance of a [PyTorch generator](https://pytorch.org/docs/stable/generated/torch.randn.html) has to be passed:
+Running the code above prints one value, but if you run it again you get a different value. What is going on here? 
+
+Every time the pipeline is run, [`torch.randn`](https://pytorch.org/docs/stable/generated/torch.randn.html) uses a different random seed to create Gaussian noise which is denoised stepwise. This leads to a different result each time it is run, which is great for diffusion pipelines since it generates a different random image each time.
+
+### CPU
```
Comment (pcuenca) [inline docs/source/en/using-diffusers/reproducibility.mdx:52]: The transition to talking about `CPU` is not yet clear at this point, as we've only set the stage for the problem. I'd maybe suggest something like this:

```suggestion

But what if you really need to generate reproducible results for your project? The solution to this problem is nuanced and depends on the hardware you use.

### CPU
```

File: docs/source/en/using-diffusers/reproducibility.mdx [hunk 2]
```diff
@@ -45,11 +45,13 @@ image = ddim(num_inference_steps=2, output_type="np").images
...
+
+But if you want to reproduce the same image on a CPU, you'll need to use a PyTorch [`Generator`](https://pytorch.org/docs/stable/generated/torch.randn.html) and set a seed:
```
Comment (pcuenca) [inline docs/source/en/using-diffusers/reproducibility.mdx:54]:
```suggestion
To generate reproducible results on a CPU, you'll need to use a PyTorch [`Generator`](https://pytorch.org/docs/stable/generated/torch.randn.html) and set a seed:
```

**Review by pcuenca — APPROVED** [review]
I love the changes! Amazing work :)

**Pull request merged**